raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -164,7 +165,7 @@ private boolean becomeFollower(int newEpoch, Function<FollowerState, Boolean> fu
 
         FollowerState followerState = followerStateOrThrow();
         if (func.apply(followerState) || stateChanged) {
-            store.write(followerState.election());
+            store.writeElectionState(followerState.election());
```

Comment:
We need to think through the exception handling a little bit. If we fail to write the election state, then potentially it should be a fatal error. At least in some cases. On the other hand, if there is no vote to record, then we can recover without violating any guarantees, though it might not be worth the effort to implement this logic.

Reply:
I guess we could fail for v1 for now to leave the file IO complexity.

Reply:
Added a ticket to track the discussion: https://confluentinc.atlassian.net/browse/KCORE-431